### PR TITLE
Single dockerfile

### PR DESCRIPTION
**Pull request by tomleb**
## Issue

(Currently includes changes from https://github.com/rancher/rancher/pull/49633)

### CODE REVIEW

**Review by pjbgf — APPROVED** [review]
LGTM - good job putting this together @tomleb. :clap:

**Comment by tomleb** [comment]
@tashima42 As discussed, I removed the dependency between `build-agent` and `build-server` in the CI for both PR and release GHA. 

Turns out the CI also becomes faster now that the agent can be built in parallel.. So we shave off maybe ~10 minutes from the CI.

**Before**

~20 minutes for building all images (not in parallel)

image


**After**

~10 minutes for building all images (in parallel)

image

I tried to have agent and server built in the same step here but that results in also slow CI. It looks like the `go build` steps is what takes the majority of time here (~8 minutes) so we don't gain much from the caching apparently :shrug: I'd still want to investigate this a bit because I'd expect some speed up from GOCACHE but anyway... (Here's the PR for that test: https://github.com/rancher/rancher/pull/49898)

File: package/Dockerfile [hunk 1]
```diff
@@ -235,61 +299,68 @@ ENV ETCD_UNSUPPORTED_ARCH ${ETCD_UNSUPPORTED_ARCH}
...
 
 # Enable exporting of the k3s images as part of the build process.
-FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.24 AS images
-
-ENV GOMODCACHE=/root/.cache/go/modcache
-ENV GOCACHE=/root/.cache/go/cache
-
+FROM --platform=$BUILDPLATFORM go-builder AS images
+WORKDIR /src
+ENV CGO_ENABLED=1
```
Comment (tashima42) [inline package/Dockerfile:304]: Shouldn't CGO be disabled here?
```suggestion
ENV CGO_ENABLED=0
```

Comment (tomleb) [reply package/Dockerfile:304]: CGO is enabled only for the airgap binary (export-images). It was enabled before and it still is, so that remain unchanged.

I did try previously to turn it off but it failed. That stage downloads some btrfs-devel and gpgme-devel libraries and I'm guess we're making use of those things via CGO.

File: package/Dockerfile [hunk 2]
```diff
@@ -235,61 +299,68 @@ ENV ETCD_UNSUPPORTED_ARCH ${ETCD_UNSUPPORTED_ARCH}
...
+RUN --mount=type=cache,target=/root/.cache,id=rancher go mod download
+RUN --mount=type=cache,target=/root/.cache,id=rancher cd pkg/apis && go mod download
+RUN --mount=type=cache,target=/root/.cache,id=rancher cd pkg/client && go mod download
+
+
+FROM --platform=$BUILDPLATFORM rancher-go-builder AS server-build
 ARG VERSION
 ARG COMMIT
 ARG RKE_VERSION
-ARG CGO_ENABLED=0
```
Comment (tomleb) [inline package/Dockerfile:260]: @tashima42 since we're on the topic of CGO, we can see here there was an `ARG CGO_ENABLED=0` but no variables used and no `ENV` directive either. So it feels like CGO was enabled before. My PR does add `ENV CGO_ENABLED=$CGO_ENABLED` though so now CGO should definitely be disabled when building the go binary.

Do you see this as an issue?

**Review by tashima42 — DISMISSED** [review]

**Comment by tashima42** [comment]
Approved for now, but we need to take a more in-depth look at `tests/v2/codecoverage/package/*` and maybe find a way to not duplicate Dockerfiles

**Review by tashima42 — APPROVED** [review]
